gammapy/cube/psf_kernel.py

```diff
@@ -0,0 +1,310 @@
+    kernel_map = kernel_map.downsample(factor, preserve_counts=False)
+
+    if normalize:
+        # normalize each image in map
+        for img, idx in kernel_map.iter_by_image():
+            norm = np.sum(img)
+            img /= norm
+
+    return kernel_map
+
+
+def energy_dependent_table_psf_to_kernel_map(table_psf, geom, normalize=True, factor=4):
+    """Compute an energy dependent PSF kernel on a given MapGeom.
+
+    The PSF is estimated by oversampling defined by a given factor.
+
+    Parameters
+    ----------
+    table_psf : `~gammapy.irf.EnergyDependentTablePSF`
+        the input table PSF
+    geom : `~gammapy.maps.MapGeom`
+        the target geometry. The PSF kernel will be centered on the spatial centre.
+        the geometry axes should contain an energy MapAxis.
+    normalize : bool
+        normalize the PSF kernel (per energy)
+    factor : int
+        the oversample factor to compute the PSF
+    """
+    # Find energy axis in geom
+    energy_idx = -1
+    for i, axis in enumerate(geom.axes):
+        if axis.type is 'energy':
+            energy_idx = i
+
+    if energy_idx == -1:
+        raise ValueError("No energy axis in target geometry for PSFKernel")
+
+    energy_unit = u.Unit(geom.axes[energy_idx].unit)
+
+    # TODO: change the logic to support non-regular geometry. This would allow energy dependent sizes for the kernel.
+    if geom.is_regular is False:
+        raise ValueError("Non regular geometries non supported yet.")
+
+    # First upsample spatial geom
+    upsampled_image_geom = geom.to_image().upsample(factor)
+
+    # get center coordinate
+    center_coord = upsampled_image_geom.center_coord * u.deg
+
+    # get coordinates
+    map_c = upsampled_image_geom.get_coord()
+
+    # get solid angles
+    solid_angles = upsampled_image_geom.solid_angle()
+
+    # compute distances to map center
+    rads = angular_separation(center_coord[0], center_coord[1], map_c.lon * u.deg, map_c.lat * u.deg)
+
+    # Create map
+    kernel_map = Map.from_geom(geom=upsampled_image_geom.to_cube(axes=geom.axes),
+                               unit='')
+
+    # loop over images
+    for img, idx in kernel_map.iter_by_image():
+        energy = geom.axes[energy_idx].center[idx[energy_idx]] * energy_unit
+        vals = table_psf.evaluate(energy=energy, rad=rads).reshape(img.shape) * solid_angles
+        img += vals.value
+
+    # downsample the psf kernel map. Take the average
+    kernel_map = kernel_map.downsample(factor, preserve_counts=False)
+
+    if normalize:
+        # normalize each image in map
+        for img, idx in kernel_map.iter_by_image():
+            norm = np.sum(img)
+            img /= norm
+
+    return kernel_map
+
+
+class PSFKernel(object):
+    """PSF kernel for `~gammapy.maps.Map`.
+
+    This is a container class to store a PSF kernel
+    that can be used to convolve `~gammapy.maps.WcsNDMap` objects.
+    It is usually computed from an EnergyDependentTablePSF
```

Comment:
Suggest to either just remove this "It is usually computed from an EnergyDependentTablePSF", and mention the classmethods instead that are usually used to create a kernel. Or add an `Examples` section showing the user how to make a kernel. Having a minimal example one can copy & paste is really nice to learn a new class.
See e.g. http://docs.gammapy.org/dev/maps/index.html#constructing-with-factory-methods with nice examples how to make maps for testing and learning quickly.